### ISSUE TITLE
Plans Page: Redesign for a more effective plans page

### ISSUE THREAD

**Issue by jeffgolenski** [Automattic/jetpack#6310]
As per @richardmuscat concepts, we want to change the overall design of the plans page within the plugin to be more actionable and offer more value to customers. 

Concepts: https://cloudup.com/ctxEaF130iu

Adding https://github.com/Automattic/wp-calypso/issues/7660#issuecomment-244201519 for reference as well. 

cc @rickybanister

**Comment by jeherve** [comment]
Some more ideas in #6973 and #7287

**Comment by jeherve** [comment]
This is definitely needed. Some things are quite confusing right now. Picture the following.

I purchased a plan on my site a long time ago, but I don't really use my site anymore so I'm thinking I may not renew this year. I don't exactly remember when I purchased it though, or if it saved my credit card when I did. Right now, I would like to find out more about my purchase and either cancel my plan or make sure it doesn't get renewed for me.

In another scenario, I may just want to print a receipt. 

Let's consequently open the dashboard and find that scenario:
- No menu item to help me.
- Oh wait, there is this "Plans" tab. 
- There we go, I now know I have a Professional plan:

![screenshot 2017-09-29 at 22 43 49](https://user-images.githubusercontent.com/426388/31035277-b10ccaea-a567-11e7-8dbb-6347deb3bf9c.png)

Now what? There is no link that could help me on that page. The only helpful button is the one to contact support.

We should at least add a button leading to `me/purchases/:siteUrl/id` or just `me/purchases` on WordPress.com, with a text along the lines of "Manage your plan". We could even make this a new menu item appearing on all Jetpack dashboard pages when you have a plan.
